apollo-router/src/configuration/expansion.rs

```diff
@@ -176,12 +184,15 @@ impl Expansion {
                 default.default.clone()
             };
 
+            let config_value_from_path =
+                jsonpath_lib::select(config, &format!("$.{}", default.config_path))
+                    .unwrap_or_default();
+
             if default.env_name.is_some()
-                && !jsonpath_lib::select(config, &format!("$.{}", default.config_path))
-                    .unwrap_or_default()
-                    .is_empty()
+                && !config_value_from_path.is_empty()
+                && value == default.default
+                && value != *config_value_from_path[0]
```

Comment:
Not sure what `config_value_from_path` really has in its value. If you have any idea @BrynCooke ?

Reply:
It's whatever is coming from the config file. 

There is a problem with this module unfortunately. 
`ConfigDefault` should probably be called `ConfigOverride`. These are used to override anything that is in the config, e.g. for dev mode.
Also `ConfigDefault::default` should be `ConfigOverride::override` where `override` is an `Option<Value>`

With this in mind the logic becomes a bit clearer.
```
if env.is_some() {
 // use the env value
}
else if config_override.is_some() {
 // use config_override.override
}
else {
  continue
} 
```

I'm happy to take your branch and finish off these bits if you would like, as this is unfortunately not the minor change that I thought it was. Or if you are happy to work through this that that would also be great.

Reply:
Thank you for your help and does a pair programming session is an option? If not, I'll try something in my side following your direction and If I struggle with, I'll let you finish if you're ok.

Btw I've ended up with this solution because the default.env` is always ˋSome()` because the listen one has been declared in ConfigDefault builder

Reply:
@ptondereau Sorry it took a while to respond. It's been a busy couple of weeks.
Happy for you to continue, and could possibly pair if you overlap with EU timezones.

Just to clarify, `env.is_some()` in the psudocode should be true if `ConfigDefault::env_default` is filled and there is an env variable set.

Just going to repeat the desired behaviour for completeness:

* If there is an `env_name` for the default then check to see if the corresponding env variable is set and if it is then use it.
* Use the `ConfigDefault::default`

Reply:
I'm in GMT+2 timezone. Feel free to reach me.